james-mailets/src/main/java/net/leangen/james/mailet/sample/LogAndDoNothingMailet.java

```diff
@@ -4,17 +4,21 @@
 import lombok.extern.slf4j.Slf4j;
 import org.apache.mailet.Mail;
 import org.apache.mailet.base.GenericMailet;
+import org.apache.mailet.base.GenericMatcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-@Slf4j
```

Comment:
Why don't you use lombok? 
The code is identical to what you did and much simpler to maintain.

Reply:
I don't mind, but the class was not compiling and my progress is already slow enough that I didn't want to waste more time chasing this.

Reply:
Did you activate annotation processing ? (File -> Settings -> search for annotation .. )
Also install the plugin, it helps with intellisense https://projectlombok.org/setup/intellij .

Reply:
I recommend you watch the 4 minute demo https://projectlombok.org/ to see why lombok is awesome .

Reply:
I installed the plugin and now it is working.

Reply:
Just a (probably useless and time-wasting) comment on this issue: I think that using annotations can be great, but relying on a complex IDE setup just to make the code compile is questionable. IMO James in its entirety has too much "accidental complexity" (let alone the actual domain complexity). For starters, I am not convinced it is desirable to tie the project to IntelliJ the way it currently is. On the other hand, the project has gone so far in this direction that it seems pretty futile to even consider going back now. The more I use James, the smarter I think those OSGi fellows are. For instance, they also use annotations, but not in the "magical" way that Lombok does.

Reply:
Lombok uses the same kind of magic that OSGI uses: compile time annotations. Lombok annotations don't reach runtime.
It's like writing code, but in a way that you know exactly how it's going to be and also you don't have to read it.

Java records will will the same gap /role as Lombok @Data and I'm sure they where inspired by how lombok solved problems in this area. 

Also, this is an App, not James per se. 
And lombok is not needed on the classpath to run the app.

Reply:
I understand that there are benefits. And quite possibly the benefits are worthwhile. I simply wrote that it is questionable (meaning that even though it may be the "right" thing, it is not the "perfect" thing). I always feel more comfortable when decisions are made by not ignoring the downside, and by balancing the trade-offs using proper arguments, so I feel compelled to point out these questions and don't feel convinced if the arguments are too one-sided.

Since you mention it, though, the issue is not with compile-time annotations, but rather with the bytecode magic and the compile cycle, if I can call it that (I am not a compiler expert). I have no trouble at all with compile-time annotations. As you pointed out yourself, by using Lombok with James, I can no longer compile James without a plugin and a co-operating compilation environment. (As a comparison, with OSGi, annotations are designed so I can still compile as "normal" Java even without bnd.) That is the trade-off. Is it worth making? That is another question, but not one I am really willing to spend any time on, so if you are convinced that we need it, I won't argue. I am just making the question explicit because I think somebody has to point it out. 😀

Reply:
To be clear: I am interested in learning it. It looks like it could help with productivity and readability, even if it does make the compilation environment more complex and creates compile-time dependencies. I suspect that the tradeoff is worthwhile.

Reply:
If you watch the video, it demonstrates how to compile a java class with Lombok annotations with `javac` and the lombok on the classpath. 

Annotation processing is a feature of Java compiler since Java 1.6 ?! .
https://docs.oracle.com/javase/8/docs/api/javax/annotation/processing/Processor.html 

Glad to hear you are open to using this.
It has very few downsides in practice: a very simple setup and a small learning curve. 
The plugin installation is used for IDE to work. 
Without it, Lombok will still do it's thing but Idea will not notice and throw some errors.